### ISSUE TITLE
GoogleAdsService doesn't properly apply retries when Retry is passed in

### ISSUE THREAD

**Issue by Kache** [googleapis/gapic-generator-python#1242]
Original report: https://github.com/googleads/google-ads-python/issues/597
Excerpt:

> `GoogleAdsService` provides `search()`, which accepts a [`retry: google.api_core.retry.Retry`](https://github.com/googleapis/python-api-core/blob/c89f55d10df50c222d05ea046e07e05ef7d17725/google/api_core/retry.py#L223). The Retry decorator ends up only getting [applied on the first request](https://github.com/googleads/google-ads-python/blob/0b928474b588422f421aadf12948538378fc770f/google/ads/googleads/v10/services/services/google_ads_service/client.py#L3292-L3294), making it effectively useless for large results with many pages.
>
> Subsequent requests are issued by `SearchPager`, [initialized a few lines down](https://github.com/googleads/google-ads-python/blob/0b928474b588422f421aadf12948538378fc770f/google/ads/googleads/v10/services/services/google_ads_service/client.py#L3298-L3300), where `retry` is not passed in, thus making it unavailable [at evocation](https://github.com/googleads/google-ads-python/blob/main/google/ads/googleads/v10/services/services/google_ads_service/pagers.py#L71-L73), where every `SearchPager._method()` call is not retry-decorated.

See original report (above) for environment details, steps to reproduce, some context background, follow questions, etc.

**Comment by vchudnov-g** [comment]
We're catching up on old bug reports. Is this still an issue?

**Comment by Kache** [comment]
Yeah, probably. I've made custom fixes in my production code that's been in place for a long time now.

With the fix, search() is far more reliable than search_stream(), which doesn't even properly handle retries and timeouts in the first place, given its streaming nature and the naive retry and timeout semantics.
